app/routes.py

```diff
@@ -1,2 +1,234 @@
-from flask import Blueprint
+from re import match
+from app import db
+from flask import Blueprint, request, make_response, jsonify, abort
+from app.models.task import Task
+from app.models.goal import Goal
+from datetime import datetime
+import requests
+import os
+
+
+
+# DEFINE BLUEPRINT
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+#-----------------
+#HELPER FUNCTIONS 
+def get_task_from_id(id):
+    try:
+        id = int(id)
+    except:
+        abort(400, {"error": "invalid id"})
+    return Task.query.get_or_404(id)
+
+def get_goal_from_id(id):
+    try:
+        id = int(id)
+    except:
+        abort(400, {"error": "invalid id"})
+    return Goal.query.get_or_404(id)
+
+#-----------------
+#CREATE (aka POST)
+@tasks_bp.route("", methods=["POST"])
+def create_task():
+    request_body = request.get_json()
+    
+    if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+        return make_response(jsonify({"details" : "Invalid data"}), 400)
```

Comment:
If you're returning a **dictionary** you don't need to call `jsonify` or `make_response` Flask's default return behavior will work just fine.

```suggestion
        return {"details" : "Invalid data"}, 400
```

When you return a dictionary Flask automatically makes a JSON response.  (This isn't true for a **list** which is why we need `jsonify` sometimes.)